src/utils/requestUtils.ts

```diff
@@ -3,90 +3,49 @@
 
+import { HttpOperationResponse, RequestPrepareOptions, ServiceClient, WebResource } from "@azure/ms-rest-js";
 import * as fse from 'fs-extra';
-import { HttpMethods, ServiceClientCredentials, WebResource } from "ms-rest";
 import * as path from 'path';
-import * as requestP from 'request-promise';
-import { appendExtensionUserAgent, ISubscriptionContext, parseError } from "vscode-azureextensionui";
+import { createGenericClient, parseError } from "vscode-azureextensionui";
 import { ext } from '../extensionVariables';
 import { localize } from '../localize';
 import { getWorkspaceSetting } from "../vsCodeConfig/settings";
+import { nonNullProp } from "./nonNull";
 
 export namespace requestUtils {
-    export type Request = WebResource & requestP.RequestPromiseOptions;
-
     const timeoutKey: string = 'requestTimeout';
 
-    export async function getDefaultRequestWithTimeout(url: string, credentials?: ServiceClientCredentials, method: HttpMethods = 'GET'): Promise<Request> {
-        const request: Request = await getDefaultRequest(url, credentials, method);
+    export async function sendRequestWithTimeout(options: RequestPrepareOptions): Promise<HttpOperationResponse> {
+        let request: WebResource = new WebResource();
+        request = request.prepare(options);
+
         const timeoutSeconds: number | undefined = getWorkspaceSetting(timeoutKey);
         if (timeoutSeconds !== undefined) {
             request.timeout = timeoutSeconds * 1000;
         }
-        return request;
-    }
-
-    export async function getDefaultRequest(url: string, credentials?: ServiceClientCredentials, method: HttpMethods = 'GET'): Promise<Request> {
-        const request: WebResource = new WebResource();
-        request.url = url;
-        request.method = method;
-        request.headers = {
-            ['User-Agent']: appendExtensionUserAgent()
-        };
-
-        if (credentials) {
-            await signRequest(request, credentials);
-        }
 
-        return request;
-    }
-
-    export async function getDefaultAzureRequest(urlPath: string, context: ISubscriptionContext, method: HttpMethods = 'GET'): Promise<Request> {
-        let baseUrl: string = context.environment.resourceManagerEndpointUrl;
-        if (baseUrl.endsWith('/')) {
-            baseUrl = baseUrl.slice(0, -1);
-        }
-
-        if (!urlPath.startsWith('/')) {
-            urlPath = `/${urlPath}`;
-        }
-
-        return getDefaultRequest(baseUrl + urlPath, context.credentials, method);
-    }
-
-    export async function sendRequest(request: Request): Promise<string> {
         try {
-            return await <Thenable<string>>requestP(request).promise();
+            const client: ServiceClient = createGenericClient();
+            return await client.sendRequest(options);
         } catch (error) {
-            if (parseError(error).errorType === 'ETIMEDOUT') {
+            if (parseError(error).errorType === 'REQUEST_ABORTED_ERROR') {
```

Comment:
Why did the error type change?

Reply:
This error type is defined by the sdk and the new sdk has different errors 🤷‍♂️ Something to watch out for in all extensions when moving to the new sdk